### PR TITLE
[BugFix] support list partition when use spark load

### PR DESCRIPTION
**Pull request by blanklin030**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #18504 

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by Astralidea** [comment]
You may need to deal with the DCO.

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarRocksListPartitioner.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+    @Override
+    public int getPartition(Object var1) {
+        PartitionType partitionType = PartitionType.getByType(partitionInfo.partitionType);
+        if (partitionType == PartitionType.UNPARTITIONED) {
+            return 0;
+        }
+        if (partitionType == PartitionType.RANGE) {
+            return 0;
+        }
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarRocksListPartitioner.java:60]: not need to process range and unpartitioned type in list partitioner

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarRocksListPartitioner.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        DppColumns key = (DppColumns) var1;
+        // get the partition columns from key as partition key
+        DppColumns partitionKey = new DppColumns(key, partitionKeyIndexes);
+        for (int i = 0; i < partitionListKeys.size(); ++i) {
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarRocksListPartitioner.java:64]: better use map

list item -> partition index

File: fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarRocksRangePartitioner.java [hunk 3]
```diff
@@ -41,15 +41,19 @@ public int numPartitions() {
...
     }
 
     public int getPartition(Object var1) {
-        if (partitionInfo.partitionType != null
-                && partitionInfo.partitionType.equalsIgnoreCase(UNPARTITIONED_TYPE)) {
+        PartitionType partitionType = PartitionType.getByType(partitionInfo.partitionType);
+        if (partitionType == PartitionType.UNPARTITIONED) {
+            return 0;
+        }
+        if (partitionType == PartitionType.LIST) {
```
Comment (wyb) [inline fe/spark-dpp/src/main/java/com/starrocks/load/loadv2/dpp/StarRocksRangePartitioner.java:56]: not need to process list in range partitioner

**Review by wyb — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Review by wyb — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-3.0

**Pull request merged**